apps/artwork_purchase/client/index.coffee

```diff
@@ -2,57 +2,122 @@ Backbone = require 'backbone'
 _ = require 'underscore'
 { ARTWORK } = require('sharify').data
 Form = require '../../../components/mixins/form.coffee'
-CurrentUser = require '../../../models/current_user.coffee'
-ArtworkInquiry = require '../../../models/artwork_inquiry.coffee'
-{ formatMessage } = require '../helpers.coffee'
+User = require '../../../models/user.coffee'
+PurchaseForm = require './purchase_form.coffee'
+SignupForm = require './purchase_signup_form.coffee'
 successTemplate = ->require('../templates/success.jade') arguments...
+AuthModalView = require '../../../components/auth_modal/view.coffee'
+mediator = require '../../../lib/mediator.coffee'
+Cookies = require '../../../components/cookies/index.coffee'
 
-class PurchaseForm extends Backbone.View
-  _.extend @prototype, Form
+class PurchaseView extends Backbone.View
 
-  initialize: ({ @artwork, @user }) ->
-    @inquiry = new ArtworkInquiry
+  initialize: ({ @artwork }) ->
+    @user = User.instantiate()
+    @$button = @$ '.js-ap-summary-submit'
+
+    @purchaseForm = new PurchaseForm
+      el: @$ '.js-ap-purchase'
+      $button: @$button
+      artwork: @artwork
+      user: @user
+    if @user.isLoggedOut()
+      @signupForm = new SignupForm
+        el: @$ '.js-ap-signup'
+        $button: @$button
+        user: @user
 
   events:
-    'submit .js-ap-form'          : 'onSubmit'
-    'click .js-ap-summary-submit' : 'onSubmit'
+    'submit .js-ap-purchase form'  : 'onSubmit'
+    'submit .js-ap-signup form'    : 'onSubmit'
+    'click  .js-ap-summary-submit' : 'onSubmit'
+    'click  .js-ap-login'          : 'onLoginClick'
 
-  onSubmit: (e) ->
+  onSubmit: (e) =>
     e.preventDefault()
-    return unless @validateForm()
-    return if @formIsSubmitting()
-
-    @$('button').attr 'data-state', 'loading'
-    formData = @serializeForm()
-    message = formatMessage _.extend { @artwork, @user }, formData
-    { name } = formData
-    @inquiry.set {
-      name,
-      message,
-      artwork: @artwork.id,
-      contact_gallery: true,
-      inquiry_url: window.location.href,
-      user: @user
-    }
+    @$button.blur()
+    if @user.isLoggedIn() then @submitPurchaseForm() else @submitSignupForm()
+
+  onLoginClick: (e) =>
+    e.preventDefault()
+    @openLoginModal()
+
+  openLoginModal: (copy) ->
+    mediator.trigger 'open:auth',
+      mode: 'login'
+      redirectTo: @artwork.href + '/checkout'
+      copy: copy
+    formData = _.extend artwork_id: @artwork.id, @purchaseForm.serializeForm()
+    Cookies.set 'purchase-inquiry', JSON.stringify formData
+
+# Signup
+  # Submit
+
+  submitSignupForm: (form, options)->
```

Comment:
Nice work breaking up these methods into small and well documented/readable chunks!